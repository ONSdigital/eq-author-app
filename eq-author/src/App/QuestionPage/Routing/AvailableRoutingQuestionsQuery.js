import React from "react";
import PropTypes from "prop-types";
import { Query } from "react-apollo";
import gql from "graphql-tag";

const GET_AVAILABLE_ROUTING_QUESTIONS = gql`
  query GetAvailableRoutingQuestions($id: ID!) {
    questionPage(id: $id) {
      id
      displayName
      availableRoutingQuestions {
        id
        displayName
        section {
          id
          displayName
        }
      }
    }
  }
`;

const AvailableRoutingQuestionsQuery = ({ pageId, children }) => (
  <Query
    query={GET_AVAILABLE_ROUTING_QUESTIONS}
    variables={{ id: pageId }}
    fetchPolicy="cache-and-network"
  >
    {children}
  </Query>
);

AvailableRoutingQuestionsQuery.propTypes = {
  pageId: PropTypes.string.isRequired,
  children: PropTypes.func.isRequired
};

export default AvailableRoutingQuestionsQuery;
