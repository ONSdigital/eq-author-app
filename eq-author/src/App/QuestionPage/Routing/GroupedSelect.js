import React from "react";
import PropTypes from "prop-types";
import { isNil } from "lodash";
import Select from "components/Forms/Select";

export const textSelect = "Please selectâ€¦";

const GroupedSelect = ({ groups, onChange, value, valid, ...otherProps }) => {
  const optGroups = groups.map(group => (
    <optgroup label={group.label} key={group.id}>
      {group.options.map(option => (
        <option
          value={option.value}
          key={option.value}
          disabled={option.disabled}
        >
          {option.label}
        </option>
      ))}
    </optgroup>
  ));

  return (
    <Select
      onChange={onChange}
      value={isNil(value) ? "" : value}
      invalid={!valid}
      {...otherProps}
    >
      {isNil(value) && (
        <option disabled value="" key="please-select">
          {textSelect}
        </option>
      )}
      {optGroups}
    </Select>
  );
};

const OptionProp = PropTypes.shape({
  label: PropTypes.string.isRequired,
  value: PropTypes.string.isRequired
});

const GroupProp = PropTypes.shape({
  label: PropTypes.string.isRequired,
  id: PropTypes.string.isRequired,
  options: PropTypes.arrayOf(OptionProp).isRequired
});

GroupedSelect.propTypes = {
  groups: PropTypes.arrayOf(GroupProp).isRequired,
  onChange: PropTypes.func.isRequired,
  value: PropTypes.string,
  valid: PropTypes.bool
};

GroupedSelect.defaultProps = {
  valid: true
};

export default GroupedSelect;
