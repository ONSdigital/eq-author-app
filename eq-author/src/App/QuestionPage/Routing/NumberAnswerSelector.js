import React from "react";
import styled from "styled-components";
import { colors, radius } from "constants/theme";
import { Number, Select, Label } from "components/Forms";
import PropTypes from "prop-types";
import { CURRENCY } from "constants/answer-types";
import CustomPropTypes from "custom-prop-types";
import VisuallyHidden from "components/VisuallyHidden";

export const RoutingNumberInput = styled(Number)`
  border-radius: ${radius};
  width: 8em;
  outline: none;
  ${props =>
    props.answerType === CURRENCY &&
    `
    padding-left: 1.2em;
  `};
`;

export const ComparatorSelector = styled(Select)`
  width: auto;
`;

const Value = styled.div`
  flex: 1 1 auto;
  display: flex;
  margin-left: 1em;
  align-items: center;
  position: relative;
`;

const CurrencySymbol = styled.div`
  position: absolute;
  opacity: 0.5;
  left: 0.5em;
`;

const NumberAnswerRoutingSelector = styled.div`
  display: flex;
  align-items: center;
  border: 1px solid ${colors.lightGrey};
  margin: 1em 0;
  border-radius: ${radius};
  padding: 1em;
`;

const NumberAnswerValueSelector = ({
  condition,
  onComparatorChange,
  handleValueChange
}) => {
  const otherProps = {
    questionPageId: condition.questionPage.id
  };

  return (
    <NumberAnswerRoutingSelector>
      <VisuallyHidden>
        <Label
          htmlFor={`routing-condition-comparator-${condition.routingValue.id}`}
        >
          Operator
        </Label>
      </VisuallyHidden>
      <ComparatorSelector
        id={`routing-condition-comparator-${condition.routingValue.id}`}
        onChange={value => onComparatorChange(otherProps, value)}
        name="comparator-select"
        value={condition.comparator}
        data-test="comparator-selector"
      >
        <option value="Equal">(=) Equal to</option>
        <option value="NotEqual">(&ne;) Not equal to</option>
        <option value="GreaterThan">(&gt;) More than</option>
        <option value="LessThan">(&lt;) Less than</option>
        <option value="GreaterOrEqual">(&ge;) More than or equal to</option>
        <option value="LessOrEqual">(&le;) Less than or equal to</option>
      </ComparatorSelector>
      <Value>
        <VisuallyHidden>
          <Label
            htmlFor={`routing-condition-value-${condition.routingValue.id}`}
          >
            Value
          </Label>
        </VisuallyHidden>
        {condition.answer.type === CURRENCY && (
          <CurrencySymbol>Â£</CurrencySymbol>
        )}
        <RoutingNumberInput
          id={`routing-condition-value-${condition.routingValue.id}`}
          min={-99999999}
          max={999999999}
          placeholder="Value"
          value={condition.routingValue.numberValue}
          name={condition.routingValue.id}
          onChange={handleValueChange}
          data-test="number-value-input"
          answerType={condition.answer.type}
        />
      </Value>
    </NumberAnswerRoutingSelector>
  );
};

NumberAnswerValueSelector.propTypes = {
  condition: PropTypes.shape({
    id: PropTypes.string.isRequired,
    comparator: PropTypes.string.isRequired,
    questionPage: PropTypes.shape({
      id: PropTypes.string.isRequired,
      displayName: PropTypes.string
    }),
    answer: CustomPropTypes.answer,
    routingValue: PropTypes.shape({
      id: PropTypes.string,
      value: PropTypes.array,
      numberValue: PropTypes.number
    })
  }).isRequired,
  onComparatorChange: PropTypes.func.isRequired,
  handleValueChange: PropTypes.func.isRequired
};

export default NumberAnswerValueSelector;
