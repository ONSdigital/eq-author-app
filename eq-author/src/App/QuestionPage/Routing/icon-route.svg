<?xml version="1.0" encoding="UTF-8"?>
<svg width="32px" height="32px" viewBox="0 0 32 32" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
    <!-- Generator: Sketch 50.2 (55047) - http://www.bohemiancoding.com/sketch -->
    <title>icon-route</title>
    <desc>Created with Sketch.</desc>
    <defs></defs>
    <g id="icon-route" stroke="none" stroke-width="1" fill="none" fill-rule="evenodd">
        <polygon id="call_split---material" fill="#4A4A4A" fill-rule="nonzero" points="17.6504483 23.6277477 19.8429574 21.4353412 14.7956188 16.4110762 14.7956188 8.37225231 16.668387 8.37225231 16.668387 15.5889238 21.1675983 20.1107622 23.3601074 17.9183557 23.3601074 23.6277477"></polygon>
        <polygon id="Path" fill="#4A4A4A" fill-rule="nonzero" points="14.2360073 23.6277477 8.63989258 23.6277477 8.63989258 18.031633 10.7888006 20.180541 13.4973202 17.4720215 14.7956188 18.7703201 12.0870993 21.4788396"></polygon>
    </g>
</svg>