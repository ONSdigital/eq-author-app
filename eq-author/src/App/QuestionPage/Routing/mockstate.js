export default [
  {
    title: "1. Integer posuere erat a ante",
    displayName: "1. Integer posuere erat a ante",
    pages: [
      {
        title: "1.1 Curabitur blandit tempus porttitor",
        displayName: "1.1 Curabitur blandit tempus porttitor",
        pageType: "QuestionPage",
        options: [
          { label: "Bibendum Cras", id: "3", selected: false },
          { label: "Aenean Quam Venenatis Elit", id: "4", selected: false },
          { label: "Purus Ornare Ipsum", id: "5", selected: false },
          { label: "Nibh", id: "6", selected: false },
          { label: "Risus Mattis Magna Bibendum", id: "7", selected: false },
          {
            label: "Consectetur Ultricies Purus Elit Fermentum",
            id: "8",
            selected: false
          }
        ],
        id: "2",
        disabled: false
      },
      {
        title: "1.2 Nulla vitae elit libero, a pharetra augue.",
        displayName: "1.2 Nulla vitae elit libero, a pharetra augue.",
        pageType: "QuestionPage",
        options: [
          { label: "Yes", id: "10", selected: false },
          { label: "No", id: "11", selected: false }
        ],
        id: "9",
        disabled: false
      }
    ],
    id: "1"
  }
];
