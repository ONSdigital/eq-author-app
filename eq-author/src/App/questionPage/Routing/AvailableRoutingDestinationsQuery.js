import React from "react";
import PropTypes from "prop-types";
import { Query } from "react-apollo";
import gql from "graphql-tag";

import LogicalDestination from "graphql/fragments/logical-destination.graphql";
import QuestionPageDestination from "graphql/fragments/question-page-destination.graphql";
import SectionDestination from "graphql/fragments/section-destination.graphql";

const GET_AVAILABLE_ROUTING_DESTINATIONS = gql`
  query GetAvailableRoutingDestinations($input: QueryInput!) {
    questionPage(input: $input) {
      id
      availableRoutingDestinations {
        logicalDestinations {
          ...LogicalDestination
        }
        questionPages {
          ...QuestionPageDestination
        }
        sections {
          ...SectionDestination
        }
      }
    }
  }

  ${LogicalDestination}
  ${QuestionPageDestination}
  ${SectionDestination}
`;

const AvailableRoutingDestinationsQuery = ({ pageId, children }) => (
  <Query
    query={GET_AVAILABLE_ROUTING_DESTINATIONS}
    variables={{ input: { pageId } }}
    fetchPolicy="cache-and-network"
  >
    {children}
  </Query>
);

AvailableRoutingDestinationsQuery.propTypes = {
  pageId: PropTypes.string.isRequired,
  children: PropTypes.func.isRequired,
};

export default AvailableRoutingDestinationsQuery;
