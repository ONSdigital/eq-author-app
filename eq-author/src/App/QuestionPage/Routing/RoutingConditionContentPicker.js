import React from "react";
import PropTypes from "prop-types";
import { get } from "lodash";

import ContentPickerSelect from "App/components/ContentPickerSelect";
import { QUESTION } from "App/components/ContentPickerSelect/content-types";
import { shapePageTree } from "App/components/ContentPicker/shapeTree";

import AvailableRoutingQuestionsQuery from "App/QuestionPage/Routing/AvailableRoutingQuestionsQuery";

export const UnwrappedRoutingConditionContentPicker = ({
  data,
  path,
  ...otherProps
}) => (
  <ContentPickerSelect
    name="questionPageId"
    contentTypes={[QUESTION]}
    questionData={shapePageTree(get(data, path))}
    {...otherProps}
  />
);

UnwrappedRoutingConditionContentPicker.propTypes = {
  data: PropTypes.object, // eslint-disable-line react/forbid-prop-types
  path: PropTypes.string.isRequired
};

export default props => (
  <AvailableRoutingQuestionsQuery pageId={props.pageId}>
    {innerProps => (
      <UnwrappedRoutingConditionContentPicker {...innerProps} {...props} />
    )}
  </AvailableRoutingQuestionsQuery>
);
