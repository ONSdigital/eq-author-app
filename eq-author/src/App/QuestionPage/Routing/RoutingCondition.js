import React from "react";
import { PropTypes } from "prop-types";
import styled from "styled-components";
import CustomPropTypes from "custom-prop-types";
import { NavLink, withRouter } from "react-router-dom";
import { TransitionGroup } from "react-transition-group";
import { lowerCase, get, isNil, uniqueId } from "lodash";

import DeleteButton from "components/Buttons/DeleteButton";
import RoutingConditionContentPicker from "App/QuestionPage/Routing/RoutingConditionContentPicker";
import MultipleChoiceAnswerOptionsSelector from "App/QuestionPage/Routing/MultipleChoiceAnswerOptionsSelector";
import NumberAnswerSelector from "App/QuestionPage/Routing/NumberAnswerSelector";
import Transition from "App/QuestionPage/Routing/Transition";
import IconText from "components/IconText";
import { Grid, Column } from "components/Grid";
import { Alert, AlertTitle, AlertText } from "App/QuestionPage/Routing/Alert";

import svgPath from "App/QuestionPage/Routing/path.svg";
import svgPathEnd from "App/QuestionPage/Routing/path-end.svg";
import IconClose from "App/QuestionPage/Routing/icon-close.svg?inline";
import { RADIO, NUMBER, CURRENCY } from "constants/answer-types";
import { buildPagePath } from "utils/UrlUtils";
import isAnswerValidForRouting from "App/QuestionPage/Routing/isAnswerValidForRouting";

import routingConditionFragment from "graphql/fragments/routing-condition.graphql";

const Label = styled.label`
  width: 100%;
  display: inline-block;
  font-size: 0.9em;
  letter-spacing: 0.05em;
  font-weight: bold;
  text-align: center;
  align-self: center;
`;

const ConnectedPath = styled.div`
  position: relative;
  height: 100%;

  &::after {
    position: absolute;
    content: "";
    background: url(${({ pathEnd }) => (pathEnd ? svgPathEnd : svgPath)})
      no-repeat center center;
    background-size: auto;
    width: 100%;
    height: calc(100% - 2em);
    top: 0;
    bottom: 0;
    margin: auto;
  }
`;

const RemoveButton = styled(DeleteButton)`
  display: block;
  margin: auto;
  position: relative;
  right: 2px;
`;

const renderNoAnswer = params => (
  <Transition exit={false}>
    <Alert data-test="no-answer-msg">
      <AlertTitle>No answers have been added to this question yet.</AlertTitle>
      <AlertText>
        First, <NavLink to={buildPagePath(params)}>add an answer</NavLink> to
        continue.
      </AlertText>
    </Alert>
  </Transition>
);

const renderUnsupportedAnswer = answer => (
  <Transition key="answer" exit={false}>
    <Alert data-test="invalid-answer-type-msg">
      <AlertTitle>Routing is not available for this type of answer</AlertTitle>
      <AlertText>
        You cannot route on &lsquo;
        {lowerCase(answer.type)}
        &rsquo; answers.
      </AlertText>
    </Alert>
  </Transition>
);

const renderDeletedQuestion = () => (
  <Transition key="answer" exit={false}>
    <Alert data-test="deleted-answer-msg">
      <AlertTitle>
        The question this condition referred to has been deleted
      </AlertTitle>
      <AlertText>
        Please select a new question from the dropdown above.
      </AlertText>
    </Alert>
  </Transition>
);

const renderCannotAddAndCondition = () => (
  <Transition key="answer" exit={false}>
    <Alert data-test="and-not-valid-msg">
      <AlertTitle>
        AND condition not valid with &lsquo;radio button&rsquo; answer
      </AlertTitle>
      <AlertText>Please select a different question.</AlertText>
    </Alert>
  </Transition>
);

const renderMultipleChoiceEditor = (condition, onToggleOption) => (
  <Transition key="answer" exit={false}>
    <MultipleChoiceAnswerOptionsSelector
      condition={condition}
      onOptionSelectionChange={onToggleOption}
    />
  </Transition>
);

const renderNumberEditor = (
  condition,
  handleComparatorChange,
  handleValueChange
) => (
  <Transition key="answer" exit={false}>
    <NumberAnswerSelector
      condition={condition}
      onComparatorChange={handleComparatorChange}
      handleValueChange={handleValueChange}
    />
  </Transition>
);

export class UnwrappedRoutingCondition extends React.Component {
  static propTypes = {
    ruleId: PropTypes.string.isRequired,
    condition: PropTypes.object.isRequired, // eslint-disable-line react/forbid-prop-types
    onConditionChange: PropTypes.func.isRequired,
    onUpdateConditionValue: PropTypes.func.isRequired,
    onToggleOption: PropTypes.func.isRequired,
    onRemove: PropTypes.func,
    label: PropTypes.oneOf(["IF", "AND"]).isRequired,
    match: CustomPropTypes.match,
    canAddAndCondition: PropTypes.bool.isRequired
  };

  constructor(props) {
    super(props);

    this.id = uniqueId("RoutingCondition");

    this.value = get(props.condition, "questionPage.id", null);
  }

  static defaultProps = {
    label: "IF"
  };

  static fragments = {
    RoutingCondition: routingConditionFragment
  };

  handleRemove = this.props.onRemove
    ? () => this.props.onRemove(this.props.ruleId, this.props.condition.id)
    : null;

  handlePageChange = ({ value: { id: questionPageId } }) => {
    this.props.onConditionChange({
      id: this.props.condition.id,
      questionPageId
    });
  };

  handleComparatorChange = (otherProps, { value }) =>
    this.props.onConditionChange({
      id: this.props.condition.id,
      questionPageId: otherProps.questionPageId,
      comparator: value
    });

  handleValueChange = ({ value, name }) =>
    this.props.onUpdateConditionValue({ id: name, customNumber: value });

  renderEditor(condition) {
    let routingEditor;
    let isValid = true;
    if (isNil(condition.questionPage)) {
      isValid = false;
      routingEditor = renderDeletedQuestion();
    } else if (isNil(condition.answer)) {
      this.value = null;
      isValid = false;
      routingEditor = renderNoAnswer(this.props.match.params);
    } else if (!isAnswerValidForRouting(condition.answer)) {
      this.value = null;
      isValid = false;
      routingEditor = renderUnsupportedAnswer(condition.answer);
    } else if (!this.props.canAddAndCondition) {
      isValid = false;
      routingEditor = renderCannotAddAndCondition();
    } else {
      isValid = true;
      this.value = get(this.props.condition, "questionPage.id", null);
      const type = get(condition, "answer.type");
      if (type === RADIO) {
        routingEditor = renderMultipleChoiceEditor(
          condition,
          this.props.onToggleOption
        );
      } else if ([CURRENCY, NUMBER].includes(type)) {
        routingEditor = renderNumberEditor(
          condition,
          this.handleComparatorChange,
          this.handleValueChange
        );
      }
    }

    return {
      routingEditor,
      isValid
    };
  }

  render() {
    const { routingEditor } = this.renderEditor(this.props.condition);

    return (
      <div data-test="routing-condition">
        <Grid align="center">
          <Column gutters={false} cols={1}>
            <Label htmlFor={this.id}>{this.props.label}</Label>
          </Column>
          <Column gutters={false} cols={10}>
            <RoutingConditionContentPicker
              id={this.id}
              pageId={this.props.match.params.pageId}
              path="questionPage.availableRoutingQuestions"
              selectedContentDisplayName={get(
                this.props,
                "condition.questionPage.displayName"
              )}
              onSubmit={this.handlePageChange}
            />
          </Column>
          <Column gutters={false} cols={1}>
            <RemoveButton
              onClick={this.handleRemove}
              disabled={!this.props.onRemove}
              data-test="btn-remove"
            >
              <IconText icon={IconClose} hideText>
                Remove
              </IconText>
            </RemoveButton>
          </Column>
        </Grid>
        <Grid>
          <Column gutters={false} cols={1}>
            <ConnectedPath pathEnd={isNil(this.props.condition.answer)} />
          </Column>
          <Column gutters={false} cols={10}>
            <TransitionGroup>{routingEditor}</TransitionGroup>
          </Column>
          <Column cols={1} />
        </Grid>
      </div>
    );
  }
}

export default withRouter(UnwrappedRoutingCondition);
