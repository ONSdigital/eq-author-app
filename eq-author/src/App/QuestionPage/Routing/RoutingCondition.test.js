import React from "react";
import { shallow } from "enzyme";
import NumberAnswerValueSelector from "App/QuestionPage/Routing/NumberAnswerSelector";
import MultipleChoiceAnswerOptionsSelector from "App/QuestionPage/Routing/MultipleChoiceAnswerOptionsSelector";
import RoutingConditionContentPicker from "App/QuestionPage/Routing/RoutingConditionContentPicker";
import { UnwrappedRoutingCondition } from "App/QuestionPage/Routing/RoutingCondition";
import sections from "App/QuestionPage/Routing/mockstate";
import { byTestAttr } from "tests/utils/selectors";

import { RADIO, TEXTFIELD, CURRENCY, NUMBER } from "constants/answer-types";

let mockHandlers, condition, match;

describe("components/RoutingCondition", () => {
  beforeEach(() => {
    mockHandlers = {
      onConditionChange: jest.fn(),
      onRemove: jest.fn(),
      onToggleOption: jest.fn(),
      onUpdateConditionValue: jest.fn()
    };

    condition = {
      id: "1",
      questionPage: {
        id: "2",
        displayName: "foobar"
      },
      answer: {
        id: "3",
        type: "Radio"
      },
      comparator: "Equal"
    };

    match = {
      params: {
        questionnaireId: "1",
        sectionId: "2",
        pageId: "3"
      }
    };
  });

  const createWrapper = (props = {}, render = shallow) =>
    render(
      <UnwrappedRoutingCondition
        id="test"
        ruleId="1"
        sections={sections}
        condition={condition}
        canAddAndCondition
        match={match}
        {...mockHandlers}
        {...props}
      />
    );

  it("should render consistently", () => {
    const wrapper = createWrapper();
    expect(wrapper).toMatchSnapshot();
  });

  it("should render message if question deleted", () => {
    condition.questionPage = null;
    const wrapper = createWrapper();
    expect(wrapper.find(byTestAttr("deleted-answer-msg")).exists()).toBe(true);
  });

  it("should render message if no answer available", () => {
    condition.answer = null;
    const wrapper = createWrapper();
    expect(wrapper.find(byTestAttr("no-answer-msg")).exists()).toBe(true);
  });

  it("should render message if invalid answer type", () => {
    condition.answer.type = TEXTFIELD;
    const wrapper = createWrapper();
    expect(wrapper.find(byTestAttr("invalid-answer-type-msg")).exists()).toBe(
      true
    );
  });

  it("should render message if cannot add and condition", () => {
    const wrapper = createWrapper({
      canAddAndCondition: false
    });
    expect(wrapper.find(byTestAttr("and-not-valid-msg")).exists()).toBe(true);
  });

  it("should render multiple choice editor correctly", () => {
    condition.answer.type = RADIO;
    const wrapper = createWrapper();
    expect(wrapper.find(MultipleChoiceAnswerOptionsSelector).exists()).toBe(
      true
    );
  });

  it("should render number editor correctly", () => {
    const answerTypes = [CURRENCY, NUMBER];
    answerTypes.forEach(answerType => {
      condition.answer.type = answerType;
      const wrapper = createWrapper();
      expect(wrapper.find(NumberAnswerValueSelector).exists()).toBe(true);
    });
  });

  it("should call onRemove handler when remove button is clicked", () => {
    const wrapper = createWrapper();
    wrapper.find(byTestAttr("btn-remove")).simulate("click");
    expect(mockHandlers.onRemove).toHaveBeenCalled();
  });

  it("should correctly submit from RoutingConditionContentPicker", () => {
    const wrapper = createWrapper();

    wrapper.find(RoutingConditionContentPicker).simulate("submit", {
      value: {
        id: "999"
      }
    });
    expect(mockHandlers.onConditionChange).toHaveBeenCalledWith({
      id: condition.id,
      questionPageId: "999"
    });
  });
});
