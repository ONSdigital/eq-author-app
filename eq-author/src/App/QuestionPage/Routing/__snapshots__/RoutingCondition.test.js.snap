// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/RoutingCondition should render consistently 1`] = `
<div
  data-test="routing-condition"
>
  <Grid
    align="center"
    fillHeight={true}
  >
    <Column
      cols={1}
      gutters={false}
    >
      <RoutingCondition__Label
        htmlFor="RoutingCondition1"
      >
        IF
      </RoutingCondition__Label>
    </Column>
    <Column
      cols={10}
      gutters={false}
    >
      <_default
        id="RoutingCondition1"
        onSubmit={[Function]}
        pageId="3"
        path="questionPage.availableRoutingQuestions"
        selectedContentDisplayName="foobar"
      />
    </Column>
    <Column
      cols={1}
      gutters={false}
    >
      <RoutingCondition__RemoveButton
        data-test="btn-remove"
        disabled={false}
        onClick={[Function]}
      >
        <IconText
          hideText={true}
          icon={[Function]}
        >
          Remove
        </IconText>
      </RoutingCondition__RemoveButton>
    </Column>
  </Grid>
  <Grid
    align="top"
    fillHeight={true}
  >
    <Column
      cols={1}
      gutters={false}
    >
      <RoutingCondition__ConnectedPath
        pathEnd={false}
      />
    </Column>
    <Column
      cols={10}
      gutters={false}
    >
      <TransitionGroup
        childFactory={[Function]}
        component="div"
      >
        <Transition__RoutingComponentTransition
          exit={false}
          key="answer"
          timeout={200}
        >
          <MultipleChoiceAnswerOptionsSelector
            condition={
              Object {
                "answer": Object {
                  "id": "3",
                  "type": "Radio",
                },
                "comparator": "Equal",
                "id": "1",
                "questionPage": Object {
                  "displayName": "foobar",
                  "id": "2",
                },
              }
            }
            onOptionSelectionChange={[MockFunction]}
          />
        </Transition__RoutingComponentTransition>
      </TransitionGroup>
    </Column>
    <Column
      cols={1}
    />
  </Grid>
</div>
`;
